### PR TITLE
Added flipping functions

### PR DESCRIPTION
**Pull request by antonwolfy**
The PR proposes to implement `dpnp.flip`, `dpnp.fliplr` and `dpnp.flipud` functions.
The `dpnp.flip` is leveraging on dpctl.tensor implementation of `flip`, while others are implemented through indexing functionality.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: tests/test_flipping.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        "dtype", get_all_dtypes(no_bool=True, no_none=True)
+    )
+    def test_lr_equivalent(self, dtype):
+        dp_a = dpnp.arange(4, dtype=dtype)
+        dp_a = dp_a[:, dpnp.newaxis] + dp_a[dpnp.newaxis, :]
+        assert_equal(dpnp.flip(dp_a, 1), dpnp.fliplr(dp_a))
+
+        np_a = numpy.arange(4, dtype=dtype)
+        np_a = numpy.add.outer(np_a, np_a)
+        assert_equal(dpnp.flip(dp_a, 1), numpy.flip(dp_a, 1))
```
Comment (vtavana) [inline tests/test_flipping.py:68]:
```suggestion
        assert_equal(dpnp.flip(dp_a, 1), numpy.flip(np_a, 1))
```

File: tests/test_flipping.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        "dtype", get_all_dtypes(no_bool=True, no_none=True)
+    )
+    def test_ud_equivalent(self, dtype):
+        dp_a = dpnp.arange(4, dtype=dtype)
+        dp_a = dp_a[:, dpnp.newaxis] + dp_a[dpnp.newaxis, :]
+        assert_equal(dpnp.flip(dp_a, 0), dpnp.flipud(dp_a))
+
+        np_a = numpy.arange(4, dtype=dtype)
+        np_a = numpy.add.outer(np_a, np_a)
+        assert_equal(dpnp.flip(dp_a, 0), numpy.flip(dp_a, 0))
```
Comment (vtavana) [inline tests/test_flipping.py:80]:
```suggestion
        assert_equal(dpnp.flip(dp_a, 0), numpy.flip(np_a, 0))
```

File: tests/test_flipping.py (hunk 1 above)
Comment (antonwolfy) [reply tests/test_flipping.py:68]: good catch! thank you!

**Review by vtavana — APPROVED** [review]
Thank you @antonwolfy

**Pull request merged**